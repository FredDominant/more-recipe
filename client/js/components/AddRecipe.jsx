import React from 'react';
import { connect } from 'react-redux';

import uploadImage from '../utils/uploadImage';
import recipeValidator from '../validation/recipeValidator';
/**
 * @description componet holds form to add recipes
 *
 * @class AddRecipe
 * @extends {React.Component}
 */
class AddRecipe extends React.Component {
/**
 * Creates an instance of AddRecipe.
 * @param {any} props
 * @memberof AddRecipe
 */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      ingredients: '',
      directions: '',
      recipeImage: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
  }
  /**
   *
   * @returns {null} null
   * @param {any} event
   * @memberof AddRecipe
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  onUpload(event) {
    const image = event.target.files[0];
    this.setState({ recipeImage: image });
  }
  /**
 *
 * @returns {null} null
 * @memberof AddRecipe
 */
  isValid() {
    const { errors, isValid } = recipeValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
   *
   * @returns {null} null
   * @memberof AddRecipe
   */
  addRecipe() {
    uploadImage(this.state.recipeImage)
      .then((response) => {
        // console.log('image url after axios is', response.data.secure_url);
        this.setState({ recipeImage: response.data.secure_url });
        // console.log('image url after updating state is', this.state.recipeImage);
        // dispatch this to store;
      })
      .catch((error) => {
        console.log(error);
      });
  }
  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  handleSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.addRecipe();
    }
  }
  /**
   *
   * @returns {HTML} html
   * @memberof AddRecipe
   */
  render() {
    const errors = this.state.errors;
    return (
      <div>
        <div className="modal fade" id="addRecipe" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
          <div className="modal-dialog" role="document">
            <div className="modal-content container">
              <div className="modal-header">
                <h5 className="modal-title"><span id="title"><h3 title="More Recipes and cooking tips">More Recipes</h3></span></h5>
                <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <div className="modal-body">
                <div className="container error-body">
                  {errors.name && <div className="alert alert-dismissible alert-danger" role="alert">{errors.name}</div>}
                  {errors.description && <div className="alert alert-dismissible alert-danger" role="alert">{errors.description}</div>}
                  {errors.ingredients && <div className="alert alert-dismissible alert-danger" role="alert">{errors.ingredients}</div>}
                  {errors.directions && <div className="alert alert-dismissible alert-danger" role="alert">{errors.directions}</div>}
                </div>
                <form onSubmit={this.handleSubmit}>
                  <div className="form-group">
                    <label htmlFor="exampleInputEmail1">Recipe Name</label>
                    <input
                      type="text"
                      className="form-control"
                      name="name"
                      id="recipeName"
                      aria-describedby="small-recipe-name"
                      placeholder="Awesome Recipe!"
                      value={this.state.name}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="recipe-description">Recipe Description:</label>
                    <input
                      type="text"
                      className="form-control"
                      name="description"
                      id="recipe-description"
                      placeholder="Awesome sauce made out of nothing!"
                      value={this.state.description}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="cook-directions">Ingredients:</label>
                    <textarea
                      className="form-control"
                      id="cook-directions"
                      name="ingredients"
                      rows="3"
                      placeholder="Onions, tomatoes, grilled chicken, nutmeg"
                      value={this.state.ingredients}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="cook-directions">Directions to cook:</label>
                    <textarea
                      className="form-control"
                      name="directions"
                      id="cook-directionss"
                      rows="3"
                      placeholder="dice tomatoes, tear chicken, boil onions, mix with nutmeg"
                      value={this.state.directions}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="recipeImage">Recipe Image</label>
                    <input
                      type="file"
                      className="form-control-file"
                      name="file"
                      id="recipeImage"
                      onChange={this.onUpload}
                    />
                  </div>
                  <button className="btn btn-primary">Add Recipe</button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({

});
export default connect(mapStateToProps)(AddRecipe);
