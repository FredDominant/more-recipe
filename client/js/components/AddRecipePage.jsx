import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import MDSpinner from 'react-md-spinner';
import toastr from 'toastr';

import uploadImage from '../utils/uploadImage';
import recipeValidator from '../validation/recipeValidator';
import addRecipe from '../actions/addRecipe';
/**
 *
 * @class AddRecipePage
 *
 * @extends {React.Component}
 */
export class AddRecipePage extends React.Component {
/**
 * @description Creates an instance of AddRecipePage.
 *
 * @param {any} props
 *
 * @memberof AddRecipePage
 */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      ingredients: '',
      directions: '',
      isUploading: false,
      recipeImage: '/images/noImage.png',
      errors: {},
      addRecipeError: '',
      uploadImageError: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
    this.handleSelectImage = this.handleSelectImage.bind(this);
  }
  /**
 * @description component lifecycle method
 *
 * @returns {null} null
 *
 * @param {nextProp} nextProp object from store
 *
 * @memberof AddRecipe
 */
  componentWillReceiveProps(nextProp) {
    this.setState({ addRecipeError: nextProp.errorMessage });
  }

  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
*
* @returns {null} null

* @param {any} event

* @memberof AddRecipe
*/
  onUpload(event) {
    const reader = new FileReader();
    reader.onload = (e) => {
      this.setState({ recipeImage: e.target.result });
    };
    reader.readAsDataURL(event.target.files[0]);
  }
  /**
*
* @returns {null} null

* @memberof AddRecipe
*/
  isValid() {
    const { errors, isValid } = recipeValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
 *
 * @returns {null} null
 *
 * @memberof AddRecipe
 */
  addNewRecipe() {
    const { recipeImage, name, description, directions, ingredients } = this.state;
    if (recipeImage !== '/images/noImage.png') {
      this.setState({ isUploading: true });
      return uploadImage(recipeImage)
        .then((response) => {
          const recipeUrl = response.data.secure_url;
          console.log('recipe url is', recipeUrl);
          this.setState({ recipeImage: recipeUrl, isUploading: false });
          this.props.createRecipe(this.state);
        })
        .catch(() => {
          this.setState({ isUploading: false, uploadImageError: 'Could not upload image. Recipe not added' });
          toastr.options = {
            closeButton: true
          };
          toastr.error(this.state.uploadImageError);
        });
    }
    console.log('jumped to this place');
    this.props.createRecipe({ name, directions, ingredients, description });
  }
  /**
*
* @returns {null} null

* @param {any} event

* @memberof AddRecipe
*/
  handleSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.addNewRecipe();
    }
  }
  /**
   *
   * @memberof AddRecipePage
   *
   * @returns {null} null
   */
  handleSelectImage() {
    $('input[type=file]').click();
  }
  /**
   *
   * @returns {null} null
   *
   * @memberof AddRecipePage
   */
  render() {
    const { errors, recipeImage, name, description, directions, ingredients } = this.state;
    return (
      <div className="container-fluid">
        <div id="add-recipe-form" className="container">
          <form onSubmit={this.handleSubmit}>
            <div className="row">
              <div className="col-sm-4">
                <div
                  className="recipe-image"
                  id="add-recipe-image-container"
                  onClick={this.handleSelectImage}
                  role="presentation"
                >
                  <img
                    className="img-thumbnail"
                    src={recipeImage}
                    alt=""
                    srcSet=""
                    id="add-recipe-image"
                  />
                </div>
                <br />
                <input
                  style={{ display: 'none' }}
                  type="file"
                  name="file"
                  id="file-upload"
                  onChange={this.onUpload}
                />
              </div>
              <div className="col-sm-8">
                <div className="form-group">
                  <label htmlFor="recipeName">Recipe Name</label>
                  <input
                    type="text"
                    id="recipeName"
                    className="form-control"
                    name="name"
                    value={name}
                    onChange={this.onChange}
                  />
                  {errors.name && <small className="form-text text-muted">
                    <span className="error-text"> {errors.name} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeDescription">Recipe Description</label>
                  <input
                    type="text"
                    id="recipeDescription"
                    className="form-control"
                    name="description"
                    value={description}
                    onChange={this.onChange}
                  />
                  {errors.description && <small className="form-text text-muted">
                    <span className="error-text"> {errors.description} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeDirections">Recipe Directions</label>
                  <textarea
                    type="text"
                    rows="5"
                    id="recipeDirections"
                    className="form-control"
                    name="directions"
                    value={directions}
                    onChange={this.onChange}
                  />
                  {errors.directions && <small className="form-text text-muted">
                    <span className="error-text"> {errors.directions} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeingredients">Ingredients</label>
                  <textarea
                    type="text"
                    rows="5"
                    id="recipeIngredients"
                    className="form-control"
                    placeholder="Comma separated list of ingredients"
                    name="ingredients"
                    value={ingredients}
                    onChange={this.onChange}
                  />
                  {errors.ingredients && <small className="form-text text-muted">
                    <span className="error-text"> {errors.ingredients} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <button className="btn btn-primary">
                    {
                      !this.state.isUploading &&
                      <h6 className="text-center"> Add Recipe </h6>
                    }
                    {
                      this.state.isUploading &&
                      <h6 className="text-center"> Uploading Image... </h6>
                    }
                    {
                      this.props.loading &&
                      <span> <MDSpinner /></span>
                    }
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  errorMessage: state.addRecipe.addRecipeErrorMessage,
  addRecipeSuccess: state.addRecipe.addRecipeSuccess,
  loading: state.isUploading
});

const mapDispatchToProps = dispatch => ({
  createRecipe: recipe => dispatch(addRecipe(recipe))
});

AddRecipePage.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};
AddRecipePage.defaultProps = {
  errorMessage: null,
  addRecipeSuccess: null
};
export default connect(mapStateToProps, mapDispatchToProps)(AddRecipePage);

