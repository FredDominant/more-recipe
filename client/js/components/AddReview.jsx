import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import addReview from '../actions/addReview';
/**
 *
 * @class AddReview
 *
 * @extends {React.Component}
 */
export class AddReview extends React.Component {
/**
 * @description Creates an instance of AddReview.
 *
 * @param {object} props
 *
 * @memberof AddReview
 */
  constructor(props) {
    super(props);
    this.state = {
      content: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  /**
   *
   * @param {object} event
   *
   * @memberof AddReview
   *
   * @returns {null} null
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
   * @returns {null} null
   *
   * @param {object} event
   *
   * @memberof AddReview
   */
  handleSubmit(event) {
    const { content } = this.state;
    const recipeId = this.props.recipeId;
    event.preventDefault();
    if (content.trim().length) {
      this.props.postReview(content, recipeId);
      this.setState({ content: '' });
    }
  }
  /**
   *
   *
   * @memberof AddReview
   *
   * @return {ReactElement} markup
   */
  render() {
    return (
      <div>
        <div id="review-form">
          <div>
            <div>
              <form onSubmit={this.handleSubmit}>
                <input
                  type="text"
                  className="form-control"
                  name="content"
                  placeholder="Add a review..."
                  onChange={this.onChange}
                  value={this.state.content}
                  required
                />
                <br />
                <button className="btn" id="add-review-button">
                  <span><i className="fas fa-comment-alt" /> </span>
                  Post Review
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
AddReview.propTypes = {
  postReview: PropTypes.func.isRequired,
  recipeId: PropTypes.number
};
AddReview.defaultProps = {
  recipeId: 0
};
const mapDispatchToProps = dispatch => ({
  postReview: (review, recipeId) => dispatch(addReview(review, recipeId))
});
export default connect(null, mapDispatchToProps)(AddReview);

