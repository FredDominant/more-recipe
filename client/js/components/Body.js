import React from 'react';
import ReactDOM from 'react-dom';
import { Switch, Route } from 'react-router-dom';

import { Home } from './Home';
import { Profile } from './Profile';
import { Recipe } from './Recipe'
import { UserHome } from './UserHome';

export class Body extends React.Component {
  render() {
    return (
			<Switch>
				<Route path='/' exact component={Home} />
				<Route path='/home' exact component={UserHome} />
				<Route path='/recipe' component={Recipe} />
				<Route path='/profile' component={Profile} />
			</Switch>
    );
  }
}
