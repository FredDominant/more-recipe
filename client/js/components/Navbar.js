import React from 'react';
import ReactDOM from 'react-dom';

/**
 * 
 * 
 * @class Navbar
 * @extends {React.Component}
 */
export class Navbar extends React.Component {
  render() {
    return(
			<div className="container-fluid">
				<nav className="row navbar navbar-expand-lg navbar-light bg-light">
						<div className="col-sm-4">
							<div className="collapse navbar-collapse" id="navbarSupportedContent">
								<ul className="navbar-nav mr-auto">
									<li className="nav-item active">
										<a className="nav-link" href="#" title="More Recipes home"><i className="fa fa-home fa-lg" aria-hidden="true"></i> <span>Recipes  |</span></a>
									</li>
									<li className="nav-item active" data-toggle="modal" data-target="#register">
										<a className="nav-link" href="#" title="Create More Recipes account"><i className="fa fa-user-plus fa-lg" aria-hidden="true"></i> <span> Register |</span></a>
									</li>
									<li className="nav-item active">
										<a className="nav-link" href="#" title="Log in to your More Recipes account"><i className="fa fa-user-circle fa-lg" aria-hidden="true"></i> <span> Log in |</span></a>
									</li>
								</ul>
							</div>
						</div>
						<div className="col-sm-4">
						</div>
						<div className="col-sm-4 icons">
							<div className="collapse navbar-collapse" id="navbarSupportedContent">
								<ul className="navbar-nav mr-auto">
									<li className="nav-item active">
										<a className="nav-link" href="#"><i class="fa fa-facebook-official fa-2x" aria-hidden="true"></i></a>
									</li>
									<li className="nav-item active">
										<a className="nav-link" href="#"><i class="fa fa-twitter fa-2x" aria-hidden="true"></i></a>
									</li>
									<li className="nav-item active">
										<a className="nav-link" href="#"><i class="fa fa-instagram fa-2x" aria-hidden="true"></i></a>
									</li>
									<li className="nav-item active">
										<a className="nav-link" href="#"><i class="fa fa-youtube-play fa-2x" aria-hidden="true"></i></a>
									</li>
									<li className="nav-item active">
										<a className="nav-link" href="#"><i class="fa fa-pinterest-p fa-2x" aria-hidden="true"></i></a>
									</li>
								</ul>
							</div>
						</div>
				</nav>
			</div>
		);
  }
}
