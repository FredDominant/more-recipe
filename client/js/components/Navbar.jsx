import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Login from '../components/Login';
import Signup from '../components/Signup';
import logOutUser from '../actions/logoutUser';
/**
 *
 * @class Navbar
 *
 * @extends {React.Component}
 */
export class Navbar extends React.Component {
  /**
 * @description Creates an instance of Navbar.
 *
 * @param {any} props
 *
 * @memberof Navbar
 */
  constructor(props) {
    super(props);
    this.state = {};
    this.handleLogout = this.handleLogout.bind(this);
  }

  /**
   * @description this method handles user logout
   *
   * @returns {function} function
   *
   * @param {any} event
   *
   * @memberof Navbar
   */
  handleLogout(event) {
    event.preventDefault();
    this.props.logOutUser();
    // this.context.router.history.push('/');
    this.props.history.push('/search');
  }
  /**
   *
   * @returns {component} react component
   *
   * @memberof Navbar
   */
  render() {
    const { firstname } = this.props;
    if (this.props.authenticated) {
      return (
        <div>
          <nav className="navbar navbar-expand-lg navbar-light bg-light">
            <div className="row" id="nav2">
              <div className="col-sm-10 col-xs-6">
                <div className="col-sm-9" id="nav-button">
                  <span id="title" className="navbar-brand">
                    <Link to="/">
                      <h3
                        className="more-recipes"
                        title="More Recipes and cooking tips"
                      >
                    More Recipes</h3></Link>
                  </span>
                  <button
                    className="navbar-toggler"
                    id="nav-toggler"
                    type="button"
                    data-toggle="collapse"
                    data-target="#navbarSupportedContent"
                    aria-controls="navbarSupportedContent"
                    aria-expanded="false"
                    aria-label="Toggle navigation"
                  >
                    <span className="navbar-toggler-icon" />
                  </button>
                </div>

              </div>
              <div className="col-sm-2 col-xs-6" id="navigation-icons">
                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                  {this.props.authenticated && <ul className="navbar-nav mr-auto">
                    <li className="dropdown nav-item active">
                      <span className="nav-link" ><span><i className="fas fa-user" /> </span>
                        <Link
                          to="#"
                          className="dropdown-toggle"
                          data-toggle="dropdown"
                          role="button"
                          aria-haspopup="true"
                          aria-expanded="false"
                        >
                          { firstname }
                          <span className="caret" />
                        </Link>
                        <ul className="dropdown-menu">
                          <div className="container" id="dropdown-items">
                            <li className="nav-item active">
                              <span className="nav-link"><span><i className="fas fa-user" /> </span>
                                <Link to="/profile">
                                Profile</Link>
                                <span className="sr-only">(current)</span></span>
                            </li>
                            <li className="nav-item active">
                              <span className="nav-link">
                                <span><i className="fas fa-utensils" /></span>
                                <Link to="/user/recipes">
                                  <span> My Recipes</span></Link>
                                <span className="sr-only">(current)</span></span>
                            </li>
                            <li className="nav-item active">
                              <span className="nav-link">
                                <span>
                                  <i className="far fa-heart" />
                                </span>
                                <Link to="/favourites">
                                  <span> Favourites</span>
                                </Link> <span className="sr-only">(current)</span>
                              </span>
                            </li>
                            <li className="nav-item active">
                              <span className="nav-link">
                                <span>
                                  <i className="fas fa-plus-circle" />
                                </span>
                                <Link to="/add-recipe"> <span> Add Recipe</span>
                                </Link> <span className="sr-only">(current)</span></span>
                            </li>
                            <hr />
                            <li className="nav-item active">
                              <Link
                                to="/"
                                className="btn btn-default"
                                onClick={this.handleLogout}
                                role="button"
                                tabIndex={0}
                              >
                                <span>
                                  <i className="fas fa-sign-out-alt" />
                                </span> Logout
                              </Link>
                            </li>
                          </div>
                        </ul>
                      </span>
                    </li>
                  </ul>}
                </div>
              </div>
            </div>
          </nav>
          {this.props.children}
        </div>
      );
    }
    return (
      <div className="">
        <nav className="navbar navbar-expand-lg navbar-light bg-light">
          <div className="row" id="nav2">
            <div className="col-sm-9" id="nav-button">
              <span id="title" className="navbar-brand">
                <Link to="/"><h3
                  className="more-recipes"
                  title="More Recipes and cooking tips"
                >More Recipes</h3></Link>
              </span>
              <button
                className="navbar-toggler"
                id="nav-toggler"
                type="button"
                data-toggle="collapse"
                data-target="#navbarSupportedContent"
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
              >
                <span className="navbar-toggler-icon" />
              </button>
            </div>
            <div className="col-sm-3" id="navigation-icons">
              <div className="collapse navbar-collapse" id="navbarSupportedContent">
                <ul className="navbar-nav mr-auto">
                  <li className="nav-item active">
                    <Link
                      to="#"
                      className="nav-link"
                      data-toggle="modal"
                      data-target="#register"
                    ><span><i className="fas fa-user-plus" /></span> REGISTER
                      <span className="sr-only">(current)</span>
                    </Link>
                  </li>
                  <li className="nav-item active">
                    <Link
                      to="#"
                      className="nav-link"
                      data-toggle="modal"
                      data-target="#login"
                    >
                      <span><i className="fas fa-sign-in-alt" /></span> LOGIN
                      <span className="sr-only">(current)</span>
                    </Link>
                  </li>
                </ul>
                <Login />
                <Signup />
              </div>
            </div>
          </div>
        </nav>
        {this.props.children}
      </div>
    );
  }
}

Navbar.propTypes = {
  logOutUser: PropTypes.func.isRequired,
  authenticated: PropTypes.bool.isRequired,
  firstname: PropTypes.string,
  children: PropTypes.node.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};
Navbar.defaultProps = {
  firstname: '',
  history: {
    push: () => {}
  }
};
const mapStateToProps = state => ({
  authenticated: state.auth.isAuthenticated,
  firstname: state.auth.user.firstname
});
export default connect(mapStateToProps, { logOutUser })(Navbar);
