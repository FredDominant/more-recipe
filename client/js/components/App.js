import React from 'react';
import ReactDOM from 'react-dom';

import { Search } from './Search';
import { Navigation } from './Navigation';
import { Body } from './Body';

 export class App extends React.Component {
  render() {
    return (
      <div>
					<Body />
      </div>
    );
  }
};
