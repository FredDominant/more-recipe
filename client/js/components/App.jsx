import React from 'react';

import Body from './Body';
import Errorboundary from '../components/Errorboundary';
import Navbar from './Navbar';

const App = () => (
  <div>
    <Navbar>
      <Errorboundary >
        <Body />
      </Errorboundary>
    </Navbar>
  </div>
);
export default App;
