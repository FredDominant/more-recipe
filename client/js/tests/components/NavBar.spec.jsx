import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import { Navbar } from '../../components/Navbar';

const props = {
  logOutUser: jest.fn(),
  authenticated: true,
  children: <div />
};
/**
 * @description test for Navbar component
 *
 * @returns {null} null
 */
function setup() {
  const wrapper = shallow(<Navbar {...props} />);

  return {
    wrapper
  };
}

describe('Test for authenticated Navbar component', () => {
  sinon.spy(Navbar.prototype, 'handleLogout');
  it('should render correctly for authenticated users', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('logOutUser() should work correctly', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn()
    };
    const logOut = wrapper.find('.btn-default');
    logOut.simulate('click', event);
    expect(Navbar.prototype.handleLogout.calledOnce).toEqual(true);
  });
});

describe('Test for non authenticated Navbar component', () => {
  it('should render correctly for unauthenticated users', () => {
    const newProps = {
      logOutUser: jest.fn(),
      authenticated: false,
      children: <div />
    };
    const wrapper = shallow(<Navbar {...newProps} />);
    expect(wrapper.find('li').length).toBe(2);
  });
});
