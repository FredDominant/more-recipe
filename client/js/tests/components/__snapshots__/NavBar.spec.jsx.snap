// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for authenticated Navbar component should render correctly for authenticated users 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="row"
      id="nav2"
    >
      <div
        className="col-sm-10 col-xs-6"
      >
        <div
          className="col-sm-9"
          id="nav-button"
        >
          <span
            className="navbar-brand"
            id="title"
          >
            <Link
              replace={false}
              to="/"
            >
              <h3
                className="more-recipes"
                title="More Recipes and cooking tips"
              >
                More Recipes
              </h3>
            </Link>
          </span>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarSupportedContent"
            data-toggle="collapse"
            id="nav-toggler"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
        </div>
      </div>
      <div
        className="col-sm-2 col-xs-6"
        id="navigation-icons"
      >
        <div
          className="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <ul
            className="navbar-nav mr-auto"
          >
            <li
              className="dropdown nav-item active"
            >
              <span
                className="nav-link"
              >
                <span>
                  <i
                    className="fas fa-user"
                  />
                   
                </span>
                <Link
                  aria-expanded="false"
                  aria-haspopup="true"
                  className="dropdown-toggle"
                  data-toggle="dropdown"
                  replace={false}
                  role="button"
                  to="#"
                >
                  <span
                    className="caret"
                  />
                </Link>
                <ul
                  className="dropdown-menu"
                >
                  <div
                    className="container"
                    id="dropdown-items"
                  >
                    <li
                      className="nav-item active"
                    >
                      <span
                        className="nav-link"
                      >
                        <span>
                          <i
                            className="fas fa-user"
                          />
                           
                        </span>
                        <Link
                          replace={false}
                          to="/profile"
                        >
                          Profile
                        </Link>
                        <span
                          className="sr-only"
                        >
                          (current)
                        </span>
                      </span>
                    </li>
                    <li
                      className="nav-item active"
                    >
                      <span
                        className="nav-link"
                      >
                        <span>
                          <i
                            className="fas fa-utensils"
                          />
                        </span>
                        <Link
                          replace={false}
                          to="/user/recipes"
                        >
                          <span>
                             My Recipes
                          </span>
                        </Link>
                        <span
                          className="sr-only"
                        >
                          (current)
                        </span>
                      </span>
                    </li>
                    <li
                      className="nav-item active"
                    >
                      <span
                        className="nav-link"
                      >
                        <span>
                          <i
                            className="far fa-heart"
                          />
                        </span>
                        <Link
                          replace={false}
                          to="/favourites"
                        >
                          <span>
                             Favourites
                          </span>
                        </Link>
                         
                        <span
                          className="sr-only"
                        >
                          (current)
                        </span>
                      </span>
                    </li>
                    <li
                      className="nav-item active"
                    >
                      <span
                        className="nav-link"
                      >
                        <span>
                          <i
                            className="fas fa-plus-circle"
                          />
                        </span>
                        <Link
                          replace={false}
                          to="/add-recipe"
                        >
                           
                          <span>
                             Add Recipe
                          </span>
                        </Link>
                         
                        <span
                          className="sr-only"
                        >
                          (current)
                        </span>
                      </span>
                    </li>
                    <hr />
                    <li
                      className="nav-item active"
                    >
                      <Link
                        className="btn btn-default"
                        onClick={[Function]}
                        replace={false}
                        role="button"
                        tabIndex={0}
                        to="/"
                      >
                        <span>
                          <i
                            className="fas fa-sign-out-alt"
                          />
                        </span>
                         Logout
                      </Link>
                    </li>
                  </div>
                </ul>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
  <div />
</div>
`;
